packaging_automation/update_package_properties.py

```diff
@@ -192,25 +237,25 @@ def update_rpm_spec(package_properties_params: PackagePropertiesParams, spec_ful
     env = get_template_environment(templates_path)
 
     rpm_version = package_properties_params.rpm_version()
-    template = env.get_template('project.spec.tmpl')
-
-    history_lines = rpm_changelog_history(spec_full_path).splitlines()
+    template = env.get_template(package_properties_params.rpm_spec_template_file_name())
+    if package_properties_params.changelog_add():
+        history_lines = rpm_changelog_history(spec_full_path).splitlines()
 
-    if len(history_lines) > 0 and package_properties_params.project_version in history_lines[1]:
-        raise ValueError(f"{package_properties_params.project_version} already exists in rpm spec file")
+        if len(history_lines) > 0 and package_properties_params.project_version in history_lines[1]:
+            raise ValueError(f"{package_properties_params.project_version} already exists in rpm spec file")
 
-    latest_changelog = convert_citus_changelog_into_rpm_changelog(package_properties_params)
-    changelog = f"{latest_changelog}\n\n{rpm_changelog_history(spec_full_path)}"
+        latest_changelog = convert_citus_changelog_into_rpm_changelog(package_properties_params)
+        changelog = f"{latest_changelog}\n\n{rpm_changelog_history(spec_full_path)}"
+    else:
+        changelog = rpm_changelog_history(spec_full_path)
     content = template.render(version=package_properties_params.project_version, rpm_version=rpm_version,
-                              project_name=package_properties_params.project_name,
+                              project_name=package_properties_params.supported_project.value.name,
                               fancy_version_no=package_properties_params.fancy_version_number, changelog=changelog)
     with open(spec_full_path, "w+") as writer:
         writer.write(content)
 
 
 def validate_package_properties_params_for_update_all_changes(package_props: PackagePropertiesParams):
-    if not package_props.project_name:
-        raise ValueError("Project Name should not be empty")
     if package_props.fancy_version_number < 0:
```

Comment:
why are we removing this check ?

Reply:
Because it is validated on main function and there is no need to validate here. 
parser.add_argument('--prj_name', choices=[r.name for r in SupportedProject])